### ISSUE TITLE
switchToTab doesn't work after 2.0.0-experimental.96

### ISSUE THREAD

**Issue by agumack** [wix/react-native-navigation#284]
Hi, switchToTab doesn't work after 2.0.0-experimental.96
Log: could not find UITabBarController

**Comment by guyca** [comment]
Seems to work fine in the example project

**Comment by agumack** [comment]
sorry @guyca 

It work on Android only.
IOS not working. "could not find UITabBarController"

**Comment by guyca** [comment]
No worries, thanks for reporting @agumack

**Comment by agumack** [comment]
Detail log:

```
2016-09-07 13:54:06.621 [warn][tid:com.facebook.react.JavaScript] Possible Unhandled Promise Rejection (id: 6):
could not find UITabBarController
createErrorFromErrorData@http://192.168.1.101:8081/index.ios.bundle?platform=ios&dev=true:7460:20
http://192.168.1.101:8081/index.ios.bundle?platform=ios&dev=true:7411:35
__invokeCallback@http://192.168.1.101:8081/index.ios.bundle?platform=ios&dev=true:7274:15
http://192.168.1.101:8081/index.ios.bundle?platform=ios&dev=true:7145:24
guard@http://192.168.1.101:8081/index.ios.bundle?platform=ios&dev=true:7072:3
invokeCallbackAndReturnFlushedQueue@http://192.168.1.101:8081/index.ios.bundle?platform=ios&dev=true:7144:6
invokeCallbackAndReturnFlushedQueue@[native code]
```

**Comment by edvinasbartkus** [comment]
I find that `navigatorID == undefined` when received in [navigatorSwitchToTab](https://github.com/wix/react-native-navigation/blob/master/src/deprecated/platformSpecificDeprecated.ios.js#L363). Therefor, it tries to find `undefined_tabs` component in iOS code which results in failure. However, I think the issue was introduced much earlier. Can't find which version exactly.

**Comment by sogko** [comment]
I encountered the same `could not find UITabBarController` exception (on iOS) recently and fix #290 did not address it in my case. 

Took awhile to track it down; finally figured out what caused my tab bar component to be missing.
Hope this helps out someone else down the line.

-----

```
- "react-native-navigation": "^2.0.0-experimental.127"
- iOS only
```

In my case:

The source of the exception was from `navigator.setTabBadge()`. I had it hooked up to an event handler that updates the badge when a value changes.

```js
onValueChanged = (size) => {
  this.props.navigator.setTabBadge({
        tabIndex: 1,
        badge: size
  });
}
```

The exception only triggers when I was dismissing a modal window.
The modal window was created from `navigator.showModal()`.

It turns out I was being overzealous when dismissing the modal when I used `navigator. dismissAllModals()`. 

This caused the RCCManager to unregister a bunch of controllers, including the tab bar controller (`TabBarControllerIOS`), probably causing `setTabBadge` not able to find the `UITabBarController`.

Using `navigator.dismissModal()` instead addresses the issue that I had and voila, no more exception.

I'm not sure if `dismissAllModals()` was misbehaving in the beginning or there's some other underlying issue, or behaviour is expected.

Cheers!
